### PR TITLE
PAE-702.3 - Update data format and use EnrollmentRequestLog.

### PR DESCRIPTION
**Pull request by Jacatove**
# Description:
This PR is intended to add the username, fullname, first name and last name in the string that is returned by `_get_enrollment_data` and also use `EnrollmentRequestLog` to persist logging information related to the enrollment.

### CODE REVIEW

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 1]
```diff
@@ -35,29 +40,48 @@ def _post_enrollment(self, data, course_settings=None):
...
             enrollment, created = ExternalEnrollment.objects.get_or_create(  # pylint: disable=no-member
                 controller_name=str(self),
                 course_shell_id=data.get('course_id'),
                 email=data.get('user_email'),
             )
         except IntegrityError:
-            LOG.error('Failed to complete enrollment, course_id and user_email can\'t be None')
+            error_msg = 'Failed to complete enrollment, course_id and user_email can\'t be None.'
+            log_details['status'] = error_msg
+            LOG.error(error_msg)
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:56]: @Jacatove, would you add a blank line before this one, please?

**Review by ivanvgh — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 2]
```diff
@@ -35,29 +40,48 @@ def _post_enrollment(self, data, course_settings=None):
         """
         LOG.info('Calling enrollment for [%s] with data: %s', self.__str__(), data)
         LOG.info('Calling enrollment for [%s] with course settings: %s', self.__str__(), course_settings)
-
+        log_details = {
+            'data': str(data),
```
[the remark this answers is not in the document]
Comment (Jacatove) [reply openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:44]: @ivanvgh Within data comes a 'CourseLocator' which is not JSON serialiazable. That is why I made the string conversion.

[the remark this answers is not in the document]
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:45]: Got it!

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py (hunk 1 above)
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:57]: @Jacatove, I meant before not after. To separate assignment to action statements.

**Review by ivanvgh — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 3]
```diff
@@ -35,29 +41,49 @@ def _post_enrollment(self, data, course_settings=None):
...
-
+            enrollment_data = self._get_enrollment_data(data)
             enrollment.meta.append(
                 {
-                    'enrollment_data_formated': self._get_enrollment_data(data),
+                    'enrollment_data_formated': enrollment_data,
                     'is_uploaded': False,
                 },
             )
+
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:72]: @Jacatove, please would you remove this unnecessary blank line?

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 4]
```diff
@@ -35,29 +41,49 @@ def _post_enrollment(self, data, course_settings=None):
...
+            LOG.error(error_msg)
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
         else:
             enrollment.meta = [] if created else enrollment.meta
-
+            enrollment_data = self._get_enrollment_data(data)
             enrollment.meta.append(
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:66]: @Jacatove, would you add a blank line before this one?

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 5]
```diff
@@ -35,29 +41,49 @@ def _post_enrollment(self, data, course_settings=None):
         """
         LOG.info('Calling enrollment for [%s] with data: %s', self.__str__(), data)
         LOG.info('Calling enrollment for [%s] with course settings: %s', self.__str__(), course_settings)
-
+        log_details = {
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:44]: @Jacatove, please add a blank line before this one.

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 6]
```diff
@@ -35,29 +41,49 @@ def _post_enrollment(self, data, course_settings=None):
...
+            'data': str(data),
+            'course_advanced_settings': course_settings,
+        }
```
Comment (ivanvgh) [inline openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:47]: @Jacatove, please add a blank line to separate the try block.

**Review by ivanvgh — CHANGES_REQUESTED** [review]

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py [hunk 7]
```diff
@@ -35,29 +40,48 @@ def _post_enrollment(self, data, course_settings=None):
...
-            LOG.error('Failed to complete enrollment, course_id and user_email can\'t be None')
+            error_msg = 'Failed to complete enrollment, course_id and user_email can\'t be None.'
+            log_details['status'] = error_msg
+            LOG.error(error_msg)
+            EnrollmentRequestLog.objects.create(  # pylint: disable=no-member
+                request_type=str(self),
+                details=log_details,
+            )
         else:
             enrollment.meta = [] if created else enrollment.meta
```
[the remark this answers is not in the document]
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:67]: @Jacatove, please address this one.

File: openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py (hunk 4 above)
Comment (ivanvgh) [reply openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py:69]: @Jacatove, please address this one.

**Review by ivanvgh — CHANGES_REQUESTED** [review]

**Review by ivanvgh — APPROVED** [review]

**Review by diegomillan — APPROVED** [review]

**Review by ivanvgh — DISMISSED** [review]

**Review by diegomillan — DISMISSED** [review]

**Review by ivanvgh — APPROVED** [review]

**Review by diegomillan — APPROVED** [review]

**Pull request merged**